src/patterns/InPlaceFilters/FilterDropdown.tsx

```diff
@@ -0,0 +1,132 @@
+import classnames from 'classnames';
+import React from 'react';
+import Button from '../../components/Button/Button';
+import { ButtonVariants } from '../../components/Button/types';
+import FilterHeader from '../../components/Filter/FilterHeader';
+import FilterInput from '../../components/Filter/FilterInput';
+import Text from '../../components/Text/Text';
+import { TextVariants } from '../../components/Text/types';
+import { px } from '../../utils';
+import { FilterDimension, FilterDropdownProps } from './types';
+import { getFilterDimensions } from './utils';
+
+export const FilterDropdown = React.forwardRef<HTMLDivElement, FilterDropdownProps>(
+  (
+    {
+      className,
+      buttonType,
+      isMobileDropdown = false,
+      filters,
+      filterIndex,
+      handleFilterSelection,
+      handleFilterUpdate,
+      clearFilterUpdate,
+      resultsCount,
+      ariaLabels,
+    },
+    ref,
+  ) => {
+    const isSortButton = buttonType === 'Sort';
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+      handleFilterSelection?.(e, buttonType);
+    };
+
+    return (
+      <div
+        className={classnames(`${px}-filter-dropdown`, className, {
+          [`${px}-filter-dropdown--mobile`]: isMobileDropdown,
+        })}
+        ref={ref}
+        data-testid={`filter-dropdown${isMobileDropdown ? '-mobile' : '-desktop'}`}
+        aria-label={
+          isMobileDropdown
+            ? ariaLabels?.dropdownMobile || `${buttonType} dropdown mobile`
```

Comment:
I don't think we need different labels here or testIds.  As for the className application above, that should be done with media queries I think not separate classnames for mobile vs desktop